### PR TITLE
Add extra metadata (as comments) to Inductor generated code

### PR DESCRIPTION
**Pull request by Chillee**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #96690
* __->__ #96581
* #96565

New output
image

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

**Comment by ezyang** [comment]
i'll let the flop counter reviewers handle this, but holler if you still need a review

**Review by ngimel — APPROVED** [review]
Cool, an example generated comment in PR description would be helpful

File: torch/_inductor/triton_ops/autotune.py [hunk 1]
```diff
@@ -233,8 +233,8 @@ def start_graph():
 def end_graph():
     if len(collected_calls) == 0:
         return
-    overall_time = sum(call[1] for call in collected_calls)
-    overall_gb = sum(call[2] for call in collected_calls)
+    overall_time = sum(call[0] for call in collected_calls)
```
[the remark this answers is not in the document]
Comment (Chillee) [reply torch/_inductor/triton_ops/autotune.py:236]: It was a bug I fixed.

File: torch/_inductor/utils.py [hunk 2]
```diff
@@ -254,13 +255,39 @@ def get_fused_kernel_name(node_schedule):
...
+    all_origins = functools.reduce(
+        operator.or_,
+        [node.node.origins for node in node_schedule if hasattr(node, "node")],
+    )
+    inductor_nodes = [origin for origin in all_origins if origin.op == "call_function"]
+    original_aten_dict = collections.defaultdict(list)
+    for node in inductor_nodes:
+        if "original_aten" in node.meta:
+            original_aten_dict[
+                str(node.meta["original_aten"]._overloadpacket)
```
Comment (shunting314) [inline torch/_inductor/utils.py:279]: Is this `original_aten` meta data initially being set by `make_fx` or somewhere else?

**Review by shunting314 — APPROVED** [review]

File: torch/_inductor/utils.py (hunk 2 above)
Comment (Chillee) [reply torch/_inductor/utils.py:279]: Yeah https://github.com/pytorch/pytorch/blob/master/torch/fx/experimental/proxy_tensor.py#L477

**Review by voznesenskym — APPROVED** [review]

**Pull request closed without merging**